internal/pkg/config/output.go

```diff
@@ -48,6 +48,12 @@ func (c *Elasticsearch) InitDefaults() {
 	c.MaxRetries = 3
 	c.MaxConnPerHost = 128
 	c.BulkFlushInterval = 250 * time.Millisecond
+
+	// Set special header "X-elastic-product-origin" for .fleet-* indices based on the latest conversation with ES team
+	// This eliminates the warning while accessing the system index
+	c.Headers = map[string]string{
+		"X-elastic-product-origin": "fleet",
+	}
```

Comment:
I think this is the incorrect to place to set it. It should not be in the configuration at the `config.Elasticsearch` level. I think it should just always be injected in the `ToESConfig` function. That function can ensure to add it to the `http.Headers`.

Reply:
sounds good, will change